File: android/src/main/java/org/ligi/gobandroid_hd/model/GameProvider.java
package org.ligi.gobandroid_hd.model;

import org.ligi.gobandroid_hd.InteractionScope;
import org.ligi.gobandroid_hd.logic.GoGame;

public class GameProvider {

    private final InteractionScope interactionScope;

    private GoGame game;

    public GameProvider(InteractionScope interactionScope) {
        this.interactionScope = interactionScope;
    }

    public GoGame get() {
        if (game == null) {
            game = new GoGame((byte) 9);
        }
        return game;
    }

    public void set(GoGame p_game) {
        interactionScope.setAsk_variant_session(true);

        game = p_game;
    }

}

File: android/src/main/java/org/ligi/gobandroid_hd/model/GameProvider.kt
package org.ligi.gobandroid_hd.model

import org.ligi.gobandroid_hd.InteractionScope
import org.ligi.gobandroid_hd.logic.GoGame

class GameProvider(private val interactionScope: InteractionScope) {

    private var game: GoGame ? = null

    fun get(): GoGame {
        if (game == null) {
            game = GoGame(9.toByte().toInt())
        }
        return game!!
    }

    fun set(p_game: GoGame) {
        interactionScope.ask_variant_session = true

        game = p_game
    }

}
